File: services/agent-engine/app/schemas/activity.py
"""Schemas for the live agent activity feed (the Office view)."""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class ActivityRead(BaseModel):
    id: int
    agent_id: str  # "openclaw:<slug>" or "builtin:<id>"
    agent_name: str
    kind: str  # chat | team | routine
    task: str
    status: str  # working | done | error
    started_at: str
    finished_at: Optional[str] = None
    note: str = ""


class ActivitySnapshotResponse(BaseModel):
    active: list[ActivityRead]
    recent: list[ActivityRead]
    generated_at: str

File: apps/desktop/src/renderer/src/lib/api/types/activity.ts
/**
 * Live agent activity (the Office view).
 *
 * Mirrors `services/agent-engine/app/schemas/activity.py`.
 */

export type ActivityKind = "chat" | "team" | "routine";
export type ActivityStatus = "working" | "done" | "error";

/** One unit of agent work — a chat turn, a team step, or a routine run. */
export interface AgentActivity {
  id: number;
  /** Namespaced id: `openclaw:<slug>` or `builtin:<db id>`. */
  agent_id: string;
  agent_name: string;
  kind: ActivityKind;
  task: string;
  status: ActivityStatus;
  started_at: string;
  finished_at: string | null;
  note: string;
}

/** Response from `GET /activity`. */
export interface ActivitySnapshot {
  active: AgentActivity[];
  recent: AgentActivity[];
  generated_at: string;
}
